### ISSUE TITLE
no-array-for-each Slower Performance

### ISSUE THREAD

**Issue by luniv04** [sindresorhus/eslint-plugin-unicorn#1522]
I went  to https://jsbench.me/ and put in the following two tests...

**Test  1:**
```js
var x = new Set([1, 2, 3, 4, 5, 6, 7, 8]);
x.forEach(value => {
	console.log(value);
});
```

**Test 2:**
```js
var x = new Set([1, 2, 3, 4, 5, 6, 7, 8]);
for(const p of x) {	
	console.log(p);
}
```

The rule "no-array-for-each" is not better to use.  The documentation indicates that using "for of" has better performance, but I dont believe it does.  I may have better performance for straight arrays, but when it comes to Maps and Sets, I am finding better performance with using the built in "forEach()" method.

**Comment by fisker** [comment]
You can disable it, if you prefer `.forEach()`.

**Comment by fisker** [comment]
Sorry,  accidentally closed it, I believe this is mistake made by #1149, this rule not mean to improve performance, but only readability.

**Comment by sindresorhus** [comment]
Micro-benchmarking code snippets like this is completely meaningless and does not reflect real-world performance. A "for-of" loop *is* faster, as it doesn't need to execute the body function for each iteration (it's just a scope). This especially matter if the loop body is complex.
